### ISSUE TITLE
Check and remedy a11y shortcomings

### ISSUE THREAD

**Issue by dhruvkb** [creativecommons/vue-vocabulary#13]
As a library of components with far-reaching scope of use, accessibility is an important aspect of the project that has unfortunately been overlooked due to the time-bound nature of the project's inception under GSoC program.

The goal is to develop a system to validate every components compliance with various accessibility guidelines. Storybooks does have an [a11y addon](https://www.npmjs.com/package/@storybook/addon-a11y) that could help with this and potentially tie in to issue #6.

References:
- [ARIA on MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)
- [W3C WAI](https://www.w3.org/WAI/)
- [WCAG on W3C](https://www.w3.org/WAI/standards-guidelines/wcag/)

**Comment by dhruvkb** [comment]
The a11y addon for Storybook has been added in 609395d.

**Comment by rashadmad** [comment]
I want to put my name down for this and am looking for clearer understanding. I have some experience with travis CI maybe we can set up a check that makes sure that future contributions are meeting a11y standards for future deployments. New documentation will need to be created to let developers know about the standards that need to be met for new changes. Its one thing to make the site assessable as it is now but we would need to continue to keep it that way for new developments. I will need to get more comfortable with the a11y addon I will keep the questions coming.

**Comment by dhruvkb** [comment]
Sure @rashadmad, you can start studying the addon and a11y standards. While Travis CI and new documentation are necessary, it would be better to address them in other issues. This issue is specifically to improve the accessibility of existing components.

**Comment by rashadmad** [comment]
So the main thing to do here is make vocabulary a11y?

**Comment by rashadmad** [comment]
I am new to Vue and some of the terms here confused me. Could you explain to me what a storybook is and can you be more descriptive about what a component is. I am already teaching my self Vue. I have experience with ruby on rails and JavaScript. I think I can get familiar with it by monday. I think that I can still do the card with my experience. Do you think that I will be alright.

**Comment by dhruvkb** [comment]
Storybook is a tool we use to visualise variations of a component. You can learn more about Storybook from their [website](https://storybook.js.org/) and docs. Our storybook is deployed [here](https://creativecommons.github.io/cc-vocabulary/storybook) and you can check it out to see what it is in action.

A component is any independently functional part of a web app such as a button, or a card for example. If you see beyond visual differences, these two are just variations of a singular `Button` component, which was designed to be customisable.
![image](https://user-images.githubusercontent.com/16580576/66092427-a6ab7900-e5a9-11e9-8e00-66f47bd38661.png)
![image](https://user-images.githubusercontent.com/16580576/66092442-af9c4a80-e5a9-11e9-8d72-8dfe805f4eb2.png)
Vocabulary is essentially just a collection of such components written in Vue. The goal of the issue to ensure that these components work just as well for people with impairments as they do for people without. You can learn more about components from the Vue [docs](https://vuejs.org/v2/guide/components.html) and visit the styleguide [here](https://creativecommons.github.io/cc-vocabulary/styleguide) to see all the components Vocabulary provides.

If you have further questions, it would be better to continue the conversation on Slack **#cc-gsoc-vocabulary** to prevent issues from becoming unwieldy.

**Comment by HAMZABoxer98** [comment]
>The goal is to develop a system to validate every components compliance with various accessibility guidelines.

1- What do you mean by *system* ? do you want to validate a11y compliance programatically ?
like this: https://www.npmjs.com/package/a11y ?

2- Some elements like Tabs doesn't meet the a11y standards.

**Comment by dhruvkb** [comment]
@HAMZABoxer98 the goal of the issue is to find and fix accessibility shortcomings in the components. None of the components right now have any a11y related considerations such as
- no ARIA attributes
- zero regard for keyboard accessibility
- no `tabindex`
- missing alt and title text

and more that even I don't know about. As of now, we have an accessibility tool built into our [storybook](https://creativecommons.github.io/cc-vocabulary/storybook) but without any components there (#6), it's not serving any purpose.

For now, let's see if we can get some simple "element" components into the playground and use the a11y plugin to find and resolve lack of a11y in them.

**Comment by dhruvkb** [comment]
A11y should be a part of the component's development lifecycle from inception to finish rather than an afterthought.
